app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt

```diff
@@ -28,44 +28,35 @@ class IndianPhoneNumberValidator : PhoneNumberValidator {
   override fun validate(number: String, type: PhoneNumberValidator.Type): Result {
     return when (type) {
       MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 10 -> LengthTooShort(10)
-          number.length > 10 -> LengthTooLong(10)
-          else -> ValidNumber
-        }
+        validateMobile(number, 10, 10, 10)
       }
       LANDLINE_OR_MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 6 -> LengthTooShort(6)
-          number.length > 12 -> LengthTooLong(12)
-          else -> ValidNumber
-        }
+        validateLandlineOrMobile(number, 6, 12)
       }
     }
   }
+
+  fun validateLandlineOrMobile(number: String, startLimit: Int, endLimit: Int): Result {
+    return when {
+      number.isBlank() -> Blank
+      number.length < startLimit -> LengthTooShort(startLimit)
+      number.length > endLimit -> LengthTooLong(endLimit)
+      else -> ValidNumber
+    }
+  }
+
+  fun validateMobile(number: String, limit: Int, smallerThanNumber: Int, greaterThanNumber: Int): Result {
+    return when {
+      number.isBlank() -> Blank
+      number.length < smallerThanNumber -> LengthTooShort(limit)
+      number.length > greaterThanNumber -> LengthTooLong(limit)
+      else -> ValidNumber
+    }
+  }
 }
 
 class EthiopianPhoneNumberValidator : PhoneNumberValidator {
   override fun validate(number: String, type: PhoneNumberValidator.Type): Result {
-    return when (type) {
-      LANDLINE_OR_MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 9 -> LengthTooShort(9)
-          number.length > 10 -> LengthTooLong(10)
-          else -> ValidNumber
-        }
-      }
-      MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 9 -> LengthTooShort(9)
-          number.length > 10 -> LengthTooLong(10)
-          else -> ValidNumber
-        }
-      }
-    }
+    return IndianPhoneNumberValidator().validateLandlineOrMobile(number, 9, 10)
```

Comment:
With this, we are introduce coupling between the Ethiopia number validator and the Indian number validator. We should keep the two classes separate since there really is no reason for the Ethiopia number validator to depend on the Indian number validator.

However, the fact that you did this leads me to believe that we can merge the two classes into one, say a `LengthBasedNumberValidator` that can take in two `IntRange` (one for `MOBILE` and another for `LANDLINE_OR_MOBILE`) and those ranges can change based on the current `Country`.